### ISSUE TITLE
[Retrospective] Use E2E test for release acceptance 

### ISSUE THREAD

**Issue by mtojek** [coder/coder#7154]
I have spent some time learning about existing e2e tests, and researching what should be done to achieve expectations noted as action items in the retrospective.

_We don’t have a dedicated QA team, so let’s make sure that we have these paths covered with automatic release acceptance tests._

**Current state:**

1. Playwright starts up the enterprise version of Coder.
2. We have 2 test cases: “List Templates” and “Sign out”.
3. In case of a failure, playwright records a clip!
4. Bad experience? https://github.com/coder/coder/issues/5554 https://github.com/coder/coder/pull/6879 

**What we want to achieve:**

Test suites:

1. Fresh install next/latest Coder release.
2. Upgrade to the last Coder release.

Test scenario per test suite:

1. Upload a template
2. List templates
3. Use template to start a workspace
4. SSH to the workspace
5. Stop workspace
6. Delete workspace

Requirements:

1. Test matrix: OS, CLI vs. web.
2. Use real PostgreSQL instance.
3. Use real CLI.
4. Developers must be able to run it locally.
5. Archive Coder server logs as artifacts. 

**MVP / Phase 1**
1. Implement test scenario for the default OS and web UI.
2. Add support for test suites.
	1. Fresh install next/latest Coder release.
	2. Upgrade to the latest Coder release.
3. Trigger e2e tests in the release workflow.

**Phase 2** 
1. Test scenario - update template and workspace:
	1. Upload a template.
	2. Start a workspace.
	3. Push the new template version.
	4. Update the workspace.
2. Perform some actions as a regular user to check RBAC.

**Phase 3**
1. Mid-scenario Coder update.

**Comment by mtojek** [comment]
Another scenario idea from @bpmct:

1. Upload a template.
2. Start a workspace.
3. Push the new template version.
4. Update the workspace.

**Comment by mtojek** [comment]
Ben posted a few harmful issues from the past. It would be great to cover them tests:

https://github.com/coder/coder/issues/6799 - root cause: new CLI introduced a regression. Reproduce: create a template, create a workspace, update coder instance, push a new template.

https://github.com/coder/coder/issues/6807 - RBAC misconfigration. Reproduce: create a workspace, with regular user try to start the workspace

https://github.com/coder/coder/pull/7034 - _git askpass stopped working for multiple weeks_, TBD

Re: Askpass, I'm not familiar with that flow. I might need to understand the context first. @bpmct Did it occur only in Windows? 

https://github.com/coder/coder/issues/6746 - blank page after the update, depending on the environment and DERP connectivity

Summing up:
1. We may want to extend scenarios with a mid-scenario Coder update.
2. It would be beneficial to perform some actions as a regular user.

**Comment by matifali** [comment]
I believe now we are at a point when we can test an end to end a new release. We can use the same approach we use for deploying PR environments. 
This currently includes,
1. A real database
2. Creating a template
3. Creating a workspace
4. Stopping the workspace
5. Upgrading the deployment with a new build.

**Comment by mtojek** [comment]
Alright, looks like we tweaked E2E tests, so this can be closed.
